src/main/java/uk/gov/justice/digital/hmpps/keyworker/config/SpringFoxConfiguration.kt

```diff
@@ -0,0 +1,97 @@
+package uk.gov.justice.digital.hmpps.keyworker.config
+
+import org.springframework.beans.BeansException
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.beans.factory.config.BeanPostProcessor
+import org.springframework.boot.info.BuildProperties
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import org.springframework.context.annotation.Import
+import org.springframework.util.ReflectionUtils
+import org.springframework.web.servlet.mvc.method.RequestMappingInfoHandlerMapping
+import springfox.bean.validators.configuration.BeanValidatorPluginsConfiguration
+import springfox.documentation.builders.PathSelectors
+import springfox.documentation.builders.RequestHandlerSelectors
+import springfox.documentation.service.ApiInfo
+import springfox.documentation.service.Contact
+import springfox.documentation.spi.DocumentationType
+import springfox.documentation.spring.web.plugins.Docket
+import springfox.documentation.spring.web.plugins.WebFluxRequestHandlerProvider
+import springfox.documentation.spring.web.plugins.WebMvcRequestHandlerProvider
+import uk.gov.justice.digital.hmpps.keyworker.controllers.KeyworkerServiceController
+import java.lang.reflect.Field
+import java.time.LocalDateTime
+import java.time.ZonedDateTime
+import java.util.Date
+import java.util.Optional
+import java.util.stream.Collectors
+
+@Configuration
+@Import(BeanValidatorPluginsConfiguration::class)
+class SpringFoxConfiguration {
+
+  @Autowired(required = false)
+  private val buildProperties: BuildProperties? = null
+
+  @Bean
+  fun api(): Docket? {
+    val apiInfo = ApiInfo(
+      "Key Worker API Documentation",
+      "API for accessing the Key Worker services.",
+      getVersion(), "", contactInfo(), "", "", emptyList()
+    )
+    val docket = Docket(DocumentationType.SWAGGER_2)
```

Comment:
So I wanted to put this to OAS_30. However there are a couple of our documentation annotations which break this (and I couldn't find any easy way to fix it - and I suspect its breaking because its SpringFox. So I'm going to detail this in the ticket to upgrade to SpringDoc - because it took a while to track down which annotation was causing the problem (@apiResponse without a response class)